### PR TITLE
Feature/add sm2 pke tests

### PR DESCRIPTION
**Pull request by peiqing6888**
# Description

Add comprehensive test suite for SM2 PKE (Public Key Encryption) implementation.

## Changes

Added new test file `pke_extended.rs` with the following test cases:

- Test vectors for different plaintext lengths (empty, 1 byte, 32 bytes, 256 bytes)
- Ciphertext tampering detection tests
- Special plaintext cases (all zeros, all ones, control characters, UTF-8)
- Encryption mode tests (C1C3C2 and C1C2C3)
- Property-based tests for encryption consistency

## Testing

All tests pass successfully. The new tests provide better coverage for:
- Edge cases
- Security properties
- Encryption modes
- Randomness properties
- Error handling

## Related Issues

N/A

## Checklist

- [x] Tests added
- [x] All tests passing
- [x] No unsafe code
- [x] Documentation comments in English

### CODE REVIEW

File: sm2/tests/pke_extended.rs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    ) {
+        let dk = create_test_key();
+        let ek = dk.encrypting_key();
+
+        // Same plaintext should decrypt correctly with different randomness
+        let cipher1 = ek.encrypt(&mut rand_core::OsRng, &plaintext1).unwrap();
+        let cipher2 = ek.encrypt(&mut rand_core::OsRng, &plaintext1).unwrap();
+
+        // Different ciphertexts for same plaintext (due to randomness)
+        prop_assert_ne!(cipher1, cipher2);
```
Comment (tarcieri) [inline sm2/tests/pke_extended.rs:107]: Looks like you need to borrow here:

```suggestion
        prop_assert_ne!(&cipher1, &cipher2);
```
